File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:

    vector<int> solution(int row){
        vector<int>ans;
        ans.emplace_back(1);
        int ele=1;
        for(int col=0;col<row;col++){
            ele=ele*(row-col);
            ele/=(col+1);
            ans.emplace_back(ele);
        }
        return ans;
    }

    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>ans;
        for(int i=0;i<numRows;i++){
            ans.emplace_back(solution(i));
        }
        return ans;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {

    List<Integer> solution(int row){
        int ele=1;
        List<Integer>ans=new ArrayList<>();
        ans.add(ele);

        for(int i=0;i<row;i++){
            ele*=(row-i);
            ele/=(i+1);
            ans.add(ele);
        }
        return ans;
    }

    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>>ans=new ArrayList<>();
        for(int i=0;i<numRows;i++){
            ans.add(solution(i));
        }
        return ans;
    }
}